dropshot/src/server.rs

```diff
@@ -518,20 +518,20 @@ impl TryFrom<&ConfigTls> for rustls::ServerConfig {
             .map_err(|err| {
                 io_error(format!("failed to load certificate: {err}"))
             })?;
-        let keys = rustls_pemfile::pkcs8_private_keys(&mut key_reader)
-            .collect::<Result<Vec<_>, _>>()
+        let keys = rustls_pemfile::private_key(&mut key_reader)
+            //.collect::<Result<Vec<_>, _>>()
```

Comment:
`rustls_pemfile::private_key` docs say "Yields the first PEM section describing a private key (of any type), or an error if a problem occurs while trying to read PEM sections."

Github isn't letting me make this suggestion for the full span, but from 521-529 I'd suggest doing:

```rust
        let private_key = rustls_pemfile::private_key(&mut key_reader)
            .map_err(|err| {
                io_error(format!("failed to load private key: {err}"))
            })?
            .ok_or_else(|| io_error("no private key found".into()))?;
```